File: convert.py
import sys
try:
    from PIL import Image
    import numpy as np
except ImportError:
    print("Installing PIL and numpy...")
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "pillow", "numpy"])
    from PIL import Image
    import numpy as np

img_path = r"c:\Users\washi\Dropbox\作業用\kotodama-match\白蛇っち.jpg"
img = Image.open(img_path).convert('L')
w, h = img.size
print(f"Original size: {w}x{h}")

frame_w = w // 3
frames = []

for i in range(3):
    left = i * frame_w
    right = left + frame_w
    crop = img.crop((left, 0, right, h))
    
    # Generate bbox
    inv_crop = Image.eval(crop, lambda x: 255 - x)
    bbox = inv_crop.getbbox()
    if bbox:
        char_crop = crop.crop(bbox)
        c_w, c_h = char_crop.size
        # Add slight padding so it doesn't touch the borders
        sq_size = int(max(c_w, c_h) * 1.2)
        sq_img = Image.new('L', (sq_size, sq_size), 255)
        offset = ((sq_size - c_w) // 2, (sq_size - c_h) // 2)
        sq_img.paste(char_crop, offset)
        
        # Resize down
        res = sq_img.resize((24, 24), Image.Resampling.LANCZOS)
    else:
        res = Image.new('L', (24, 24), 255)
    
    # Enhance contrast and threshold better
    frames.append(res)

arrays = []
for f_idx, frame in enumerate(frames):
    pixels = list(frame.getdata())
    line_strs = []
    
    threshold = 200 # Since drawing has shading, err on capturing lines
    
    for y in range(24):
        line = ""
        for x in range(24):
            val = pixels[y * 24 + x]
            if val < threshold:
                line += "█"
            else:
                line += " "
        line_strs.append(f'"{line}"')
    
    array_str = "[\n    " + ",\n    ".join(line_strs) + "\n]"
    arrays.append(array_str)

print("\n// --- ANIMATION FRAMES (COPY TO index.html) ---")
print("childA: [\n" + ",\n\n".join(arrays) + "\n],")

File: convert.cs
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.IO;

public class SpriteConverter {
    public static void Convert() {
        string imgPath = @"c:\Users\washi\Dropbox\作業用\kotodama-match\白蛇っち.jpg";
        if (!File.Exists(imgPath)) {
            Console.WriteLine("File not found: " + imgPath);
            return;
        }
        using (Bitmap img = new Bitmap(imgPath)) {
            int frameW = img.Width / 3;
            int h = img.Height;
            for (int i=0; i<3; i++) {
                Bitmap crop = img.Clone(new Rectangle(i*frameW, 0, frameW, h), img.PixelFormat);
                
                int minX = frameW, minY = h, maxX = 0, maxY = 0;
                for (int y=0; y<h; y++) {
                    for (int x=0; x<frameW; x++) {
                        Color c = crop.GetPixel(x,y);
                        int brightness = (c.R + c.G + c.B) / 3;
                        if (brightness < 240) {
                            if (x < minX) minX = x;
                            if (x > maxX) maxX = x;
                            if (y < minY) minY = y;
                            if (y > maxY) maxY = y;
                        }
                    }
                }
                
                if (minX <= maxX) {
                    int cW = maxX - minX + 1;
                    int cH = maxY - minY + 1;
                    int sqSize = Math.Max(cW, cH);
                    
                    Bitmap sq = new Bitmap(sqSize, sqSize);
                    using (Graphics g = Graphics.FromImage(sq)) {
                        g.Clear(Color.White);
                        int offX = (sqSize - cW)/2;
                        int offY = (sqSize - cH)/2;
                        g.DrawImage(crop, new Rectangle(offX, offY, cW, cH), new Rectangle(minX, minY, cW, cH), GraphicsUnit.Pixel);
                    }
                    
                    Bitmap final = new Bitmap(24, 24);
                    using (Graphics g = Graphics.FromImage(final)) {
                        g.InterpolationMode = InterpolationMode.NearestNeighbor;
                        g.DrawImage(sq, 0, 0, 24, 24);
                    }
                    
                    Console.WriteLine("[");
                    for (int y=0; y<24; y++) {
                        string line = "\"";
                        for (int x=0; x<24; x++) {
                            Color c = final.GetPixel(x,y);
                            int b = (c.R + c.G + c.B) / 3;
                            if (b < 200) line += "\u2588";
                            else line += " ";
                        }
                        line += "\"";
                        if (y < 23) line += ",";
                        Console.WriteLine("    " + line);
                    }
                    Console.WriteLine("],");
                }
            }
        }
    }
}
